src/featbin/compute-filter.cc

```diff
@@ -0,0 +1,223 @@
+                " the utterance, which mean normalized using mean computed per speaker." 
+                "If false, the utterance is mean-normalized w.r.t its local mean.");
+    po.Read(argc, argv);
+    
+    if (po.NumArgs() != 2) {
+      po.PrintUsage();
+      exit(1);
+    }
+    int32 num_done = 0, num_err = 0;
+    std::string wav_rspecifier = po.GetArg(1),
+      wspecifier = po.GetArg(2);
+    
+    BaseFloatVectorWriter writer(wspecifier);
+    if (spk2utt_rspecifier != "") { 
+      SequentialTokenVectorReader spk2utt_reader(spk2utt_rspecifier);
+      RandomAccessTableReader<WaveHolder> wav_reader(wav_rspecifier);
+      for (; !spk2utt_reader.Done(); spk2utt_reader.Next()) {
+        std::string spk = spk2utt_reader.Key();
+        const std::vector<std::string> &uttlist = spk2utt_reader.Value();
+        Vector<BaseFloat> inner_prod(lpc_order),
+          norms(lpc_order);
+        int32 num_chan = wav_reader.Value(uttlist[0]).Data().NumRows();
+        std::vector<BaseFloat> spk_sum(num_chan, 0),
+          spk_count(num_chan, 0),
+          spk_mean(num_chan, 0);
+        // Accumulate waveform sum and count over all utterance in speaker
+        // to compute mean of waveform for whole speaker over all channels.
+        if (use_mean) {
+          for (size_t i = 0; i < uttlist.size(); i++) {
+            std::string utt = uttlist[i]; 
+            const WaveData &wav_data = wav_reader.Value(utt);
+            int32 num_chan = wav_data.Data().NumRows();
+            for (int32 chan = 0; chan < num_chan; chan++) {
+              spk_sum[chan] += wav_data.Data().Row(chan).Sum();
+              spk_count[chan] += wav_data.Data().NumCols();
+            }
+          }
+          for (int32 chan = 0; chan < num_chan; chan++) 
+           spk_mean[chan] =  spk_sum[chan] / spk_count[chan];
+        }
+
+        for (size_t i = 0; i < uttlist.size(); i++) {
+          std::string utt = uttlist[i];
+          if (!wav_reader.HasKey(utt)) {
+            KALDI_WARN << "Did not find wave for utterance " << utt;
+            num_err++;
+            continue;
+          }
+          const WaveData &wav_data = wav_reader.Value(utt);
+          int32 num_chan = wav_data.Data().NumRows(), this_chan = channel;
+          KALDI_ASSERT(num_chan > 0);
+          if (channel == -1) { 
+            this_chan = 0;
+            if (num_chan != 1)
+              KALDI_WARN << "Channel not specified but you have data with "
+                         << num_chan  << " channels; defaulting to zero";
+          } else {
+            if (this_chan >= num_chan) {
+              KALDI_WARN << "File with id " << spk << " has "
+                         << num_chan << " channels but you specified channel "
+                         << channel << ", producing no output.";
+              continue;
+            }
+          }
+          SubVector<BaseFloat> waveform(wav_data.Data(), this_chan);
+          int32 corr_win_size = waveform.Dim() - lpc_order;
```

Comment:
don't call this corr_win_size since you use corr_window_size elsewhere- better be consistent